### PR TITLE
Allows to fetch bigmap with a key of type string, number or object

### PR DESCRIPTION
**Pull request by roxaneletourneau**
Replace #513 

Breaking change for `BigMapAbstraction.getMultipleValues()` and `RpcContractProvider.getBigMapKeysByID()` that will return a `MichelsonMap` instead of an object to support keys of type object (michelson pair)

Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have added unit tests
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: packages/taquito/src/contract/rpc-contract-provider.ts [hunk 1]
```diff
@@ -124,61 +124,46 @@ export class RpcContractProvider extends OperationEmitter
...
-          throw ex;
-        }
-      }
-      Object.assign(bigMapValues, {[keys[0]]: val});
+  async getBigMapKeysByID<T>(id: string, keys: Array<BigMapKeyType>, schema: Schema, block?: number): Promise<MichelsonMap<MichelsonMapKey, T | undefined>> {
+    const level = await this.getBlockForRequest(keys, block)
+    const bigMapValues = new MichelsonMap<MichelsonMapKey, T | undefined>();
+
+    // Execute batch of promises in series
+    const batchSize = 5;
```
Comment (carte7000) [inline packages/taquito/src/contract/rpc-contract-provider.ts:135]: Should that be configurable?

File: packages/taquito/src/contract/rpc-contract-provider.ts [hunk 2]
```diff
@@ -124,61 +124,46 @@ export class RpcContractProvider extends OperationEmitter
...
+    let position = 0;
+    let results: Array<(T | undefined)> = [];
+
+    while (position < keys.length) {
+      const keysBatch = keys.slice(position, position + batchSize);
+      const batch = keysBatch.map((keyToEncode) => this.getBigMapValueOrUndefined<T>(keyToEncode, id, schema, level))
```
Comment (carte7000) [inline packages/taquito/src/contract/rpc-contract-provider.ts:141]: Do we have a test for the negative case? If the key is not found we return undefined. What happens when there is an unexpected exception? Should we test that these exceptions bubble up to the caller?

File: packages/taquito/src/contract/rpc-contract-provider.ts (hunk 1 above)
Comment (roxaneletourneau) [reply packages/taquito/src/contract/rpc-contract-provider.ts]: Good idea, I made the change.

File: packages/taquito/src/contract/rpc-contract-provider.ts (hunk 2 above)
Comment (roxaneletourneau) [reply packages/taquito/src/contract/rpc-contract-provider.ts:141]: I added a test case, thank you!

**Pull request merged**